nuxeo-quota-core/src/main/java/org/nuxeo/ecm/quota/QuotaStatsUpdater.java

```diff
@@ -37,17 +37,30 @@
     /**
      * Update the statistics for the given {@code docCtx} and {@code event}. Signature was changed in 5.6 to pass the
      * Event instead of the eventName to allow the implementer to rollback the transaction if needed
-     * 
+     *
      * @param session an unrestricted {@link CoreSession} to be used
      */
     void updateStatistics(CoreSession session, DocumentEventContext docCtx, Event event);
 
     /**
      * Compute the initial statistics on the whole repository for this {@code QuotaStatsUpdater}.
-     * 
+     *
+     * @param session an unrestricted {@link CoreSession} to be used
+     * @deprecated since 10.1, use other signature
+     */
+    @Deprecated
+    default void computeInitialStatistics(CoreSession session, final QuotaStatsInitialWork currentWorker) {
```

Comment:
Why do you deprecate this API ? I think this could be useful. The second signature enforces the user to give a `path`, and the code doesn't do the same thing if `path` is `null` or equals to `/`. In the second case we add `AND ecm:path STARTSWITH '/'` which could be less efficient for DB. WDYT ?